frontend/src/__tests__/cypress/cypress/e2e/modelRegistrySettings/ModelRegistrySettings.cy.ts

```diff
@@ -0,0 +1,101 @@
+import { mockDashboardConfig, mockDscStatus } from '~/__mocks__';
+import { mockDsciStatus } from '~/__mocks__/mockDsciStatus';
+import { StackCapability, StackComponent } from '~/concepts/areas/types';
+import { modelRegistrySettings } from '~/__tests__/cypress/cypress/pages/modelRegistrySettings';
+import { pageNotfound } from '~/__tests__/cypress/cypress/pages/pageNotFound';
+import { asProductAdminUser, asProjectAdminUser } from '~/__tests__/cypress/cypress/utils/users';
+
+const setupMocksForMRSettingAccess = () => {
+  asProductAdminUser();
+  cy.interceptOdh(
+    'GET /api/config',
+    mockDashboardConfig({
+      disableModelRegistry: false,
+    }),
+  );
+  cy.interceptOdh(
+    'GET /api/dsc/status',
+    mockDscStatus({
+      installedComponents: {
+        [StackComponent.MODEL_REGISTRY]: true,
+        [StackComponent.MODEL_MESH]: true,
+      },
+    }),
+  );
+  cy.interceptOdh(
+    'GET /api/dsci/status',
+    mockDsciStatus({
+      requiredCapabilities: [StackCapability.SERVICE_MESH, StackCapability.SERVICE_MESH_AUTHZ],
+    }),
+  );
+};
+
+it('Model registry settings should not be available for non product admins', () => {
+  asProjectAdminUser();
+  modelRegistrySettings.visit(false);
+  pageNotfound.findPage().should('exist');
+  modelRegistrySettings.findNavItem().should('not.exist');
+});
+
+it('Model registry settings should be available for product admins with capabilities', () => {
+  setupMocksForMRSettingAccess();
+  // check page is accessible
+  modelRegistrySettings.visit(true);
+  // check nav item exists
+  modelRegistrySettings.findNavItem().should('exist');
+});
+
+describe('CreateModal', () => {
+  beforeEach(() => {
+    setupMocksForMRSettingAccess();
+  });
+
+  it('should display error messages when form fields are invalid and not allow submit', () => {
+    modelRegistrySettings.visit(true);
+    cy.findByText('Create model registry').click();
+    // Fill in the form with invalid data
+    cy.get('#mr-name').clear();
+    cy.get('#mr-host').clear();
+    cy.get('#mr-port').clear();
+    cy.get('#mr-username').clear();
+    cy.get('#mr-password').clear();
+    cy.get('#mr-database').clear();
+    cy.get('#mr-database').blur();
+    // Assert the submit button is disabled
+    cy.findByTestId('modal-submit-button').should('be.disabled');
+    // Assert that error messages are displayed
+    cy.findByTestId('mr-name-error')
+      .should('be.visible')
+      .contains(
+        "Must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character",
+      );
```

Comment:
Asserting static text as defined in in our react components is often a bad idea because it's a 1:1 relationship and will only fail if we change the react component.

Reply:
Ack. I plan to give these tests a second pass as part of adding more in #2829, I'll address this feedback there and request a review from you for that one when it's ready if you don't mind.

Reply:
@christianvogt looking back at this, I think it has value beyond what you're saying -- we shouldn't just assert that components render certain text for the sake of it, but this is trying to assert that the error is shown vs not shown. We could do it with a test-id instead if you prefer but imo it doesn't matter much as long as the test is asserting logic and not just content.

Reply:
My concern was the `contains` check and not so much the check that the error node was visible since this is the only available error that's shown in this node.

Reply:
@christianvogt FYI I addressed this in updates to https://github.com/opendatahub-io/odh-dashboard/pull/2829